### PR TITLE
Governance app

### PR DESCRIPTION
**Pull request by shjortConcordium**
## Purpose

Make an app that allows signing the chain updates, which was removed from the main app.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: .github/workflows/build-pipeline.yml [hunk 1]
```diff
@@ -77,7 +76,30 @@ jobs:
               with:
                 name: nanox
                 path: ./concordium_nanox.elf
-    
+
+    compile-governance-nano-s:
+        name: Compile Governance app for Nano S
```
Comment (orhoj) [inline .github/workflows/build-pipeline.yml:81]: We can also sideload for nano s plus.

**Review by orhoj — CHANGES_REQUESTED** [review]

**Comment by orhoj** [comment]
The `menu` should also be able to be shared (only the text of the UI is different). The main file is also basically duplicated.

File: governance-app/src/glyphs.c [hunk 2]
```diff
@@ -0,0 +1,108 @@
+#include "glyphs.h"
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply governance-app/src/glyphs.c:1]: They are, but I guess I added them before I ignored them, they will be removed

File: governance-app/src/globals.h [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+#include "signUpdateGasRewards.h"
+#include "signUpdateMintDistribution.h"
+#include "signUpdateProtocol.h"
+#include "signUpdateTransactionFeeDistribution.h"
+#include "signUpdateTimeParameters.h"
+#include "signUpdateCooldownParameters.h"
+#include "signUpdatePoolParameters.h"
+#include "ux.h"
+
+#define CONCORDIUM_PURPOSE   1105
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply governance-app/src/globals.h:28]: Yes but the "mess to fix" to "reduction of duplication" is too rough :upside_down_face:

**Review by orhoj — CHANGES_REQUESTED** [review]

File: governance-app/src/globals.c [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#include "os.h"
+#include "ux.h"
+
+ux_state_t G_ux;
+bolos_ux_params_t G_ux_params;
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply governance-app/src/globals.c]: I don't think we usually do this? :thinking:

File: Makefile.common [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+$(info BOLOS_ENV is not set: falling back to CLANGPATH and GCCPATH)
+endif
+ifeq ($(CLANGPATH),)
+$(info CLANGPATH is not set: clang will be used from PATH)
+endif
+ifeq ($(GCCPATH),)
+$(info GCCPATH is not set: arm-none-eabi-* will be used from PATH)
+endif
+
+COMPILE_FLAGS = -DAPPNAME=\"$(APPNAME)\"
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply Makefile.common]: I didn't realise it could be done like that. :thinking:

**Review by orhoj — APPROVED** [review]

**Pull request merged**